src/shared/components/results/SearchSuggestions.tsx

```diff
@@ -33,132 +74,29 @@ const SearchSuggestions = ({
   namespace?: Namespace;
   total?: number;
 }) => {
-  const [linesToDisplay, setLinesToDisplay] = useState(1);
-  const [termsToDisplay, setTermsToDisplay] = useState(+Infinity);
-  const ref = useRef<HTMLElement>(null);
-
   // We try to not have a request if not needed, under these conditions:
-  const shouldSuggest =
-    // Only when there are results
+  const validQueryWithContent = // Only when there are results
     // also serves to delay the requests below to prioritise getting the results
     total &&
     // Only for UniProtKB
     namespace === Namespace.uniprotkb &&
     // Only for queries with content
-    !!query?.length &&
-    // And only simple queries
-    simpleQuery.test(query);
-
-  const { data } = useDataApi<MatchedFieldsResponse>(
-    shouldSuggest
-      ? `${apiUrls.search(Namespace.uniprotkb)}?${qs.stringify({
-          size: 0,
-          query,
-          showSingleTermMatchedFields: true,
-        })}`
-      : null
-  );
-
-  // Data to enrich the suggestions with nice labels
-  const { data: searchTermsData } = useDataApi<SearchTermType[]>(
-    shouldSuggest && data?.matchedFields?.length
-      ? apiUrls.queryBuilderTerms(Namespace.uniprotkb)
-      : undefined
-  );
-
-  const searchTerms = useMemo(() => {
-    if (!searchTermsData || !data?.matchedFields?.length) {
-      return null;
-    }
-    const flattenedTerms = flatten(searchTermsData);
+    !!query?.length;
 
+  if (validQueryWithContent) {
```

Comment:
I know there's currently no overlap in these conditions, so there's no case where 2 of them might be displayed, but thinking in the future where we'd want to suggest more things there might be cases where we want to have an order of priority of suggestions.
I'm suggesting this change:
```ts
if (validQueryContent) {
  if (simpleQuery.test(query)) {
    return <AdvancedSearchSuggestion query={query} />;
  }
  if (exactMatchSearchTerms.some((term) => query.includes(term))) {
    return <ExactFieldSuggestion query={query} />;
  }
  if (taxonHierarchySearchTerms.some((term) => query.includes(term)) {
    return <TaxonomyLevelsSuggestion query={query} />;
  }
  // Add more suggestions in the future here
}
return null;
```

The specific case in my mind (for another jira and PR, not to add here), is for example a query like `(mass:[* TO 50])` we might want to suggest something like "do you want to display the "Mass" column?" with a shortcut to add that column if we see that the search field and the column name match. In the case of `taxonomy` we'd had an overlap and we might not want to add both suggestions as it takes too much space